File: maximum-depth-of-binary-tree-104/Solution.java

/**
 * LeetCode problem # 104, 'Maximum Depth of Binary Tree'
 * https://leetcode.com/problems/maximum-depth-of-binary-tree
 */

import java.util.*;

class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;

    TreeNode() {
    }

    TreeNode(int val) {
        this.val = val;
    }

    TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}

class Solution {
    public int maxDepth(TreeNode root) {
        return root == null ? 0 : Math.max(maxDepth(root.left), maxDepth(root.right)) + 1;
    }

    public int maxDepthIterative(TreeNode root) {
        Queue<TreeNode> q = new LinkedList<>();
        q.add(root);
        int res = 0;

        while (!q.isEmpty()) {
            int size = q.size();
            ++res;
            for (int i = 0; i < size; ++i) {
                TreeNode curr = q.remove();
                if (curr != null) {
                    q.add(curr.left);
                    q.add(curr.right);
                }
            }
        }
        return res - 1;
    }
}

File: maximum-depth-of-binary-tree-104/solution.js
/**
 * LeetCode problem # 104, 'Maximum Depth of Binary Tree'
 * https://leetcode.com/problems/maximum-depth-of-binary-tree
 */

/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
const maxDepth = function (root) {
  if (root == null) {
    return 0;
  }
  if (root.left == null && root.right == null) {
    return 1;
  }

  let res = Number.MIN_SAFE_INTEGER;
  if (root.left != null) {
    res = Math.max(res, maxDepth(root.left));
  }
  if (root.right != null) {
    res = Math.max(res, maxDepth(root.right));
  }
  return res + 1;
};
